Docs/User Guide.md

```diff
@@ -0,0 +1,180 @@
+# VeriStand Communications Bus Template User Guide
+
+The template provided in this repo is intended to be used for creating VeriStand custom devices that implement communication protocols. This guide aims to provide some theory on the template design details on how to use the template to create a fully functional custom device.
+
+## Overview
+
+The main goal of this template is to enable custom device authors to implement the configuration and messaging for a given protocol only once. The protocol and hardware configurations are implemented separately to more easily enable supporting multiple vendors or hardware devices in a new custom device without having to start everything from a blank custom device template.
```

Comment:
>to more easily enable supporting multiple vendors or hardware devices in a new custom device without having to start everything from a blank custom device template.

to support different vendors and hardware devices in new custom devices without starting each from an empty custom device template.